[v14] Discovery: Add TAG sync configuration parameters

tigrato opened the pull request:
Backport #37914 to branch/v14

tigrato committed:
Discovery: Add TAG sync configuration parameters (#37914)

This PR allows discovery service to receive TAG Cloud matchers
parameters.

Part of https://github.com/gravitational/access-graph/issues/458

Signed-off-by: Tiago Silva <tiago.silva@goteleport.com>

gzdunek reviewed (approved):
The title should say [v14], no?

tigrato commented:
> The title should say [v14], no?

Correct. Updated

tigrato committed:
Discovery: Allow discovery service to push resources to Access Graph (#37927) (#39186)

This PR allows Teleport Discovery service to dial to Access Graph
service using its credentials signed by HostCA and configures it to poll
and push AWS resources into Access Graph service.

Part of https://github.com/gravitational/access-graph/issues/458

Signed-off-by: Tiago Silva <tiago.silva@goteleport.com>
